product_search_by_display_name/models/product_template.py

```diff
@@ -0,0 +1,18 @@
+# Copyright 2023 Komit - Cuong Nguyen Mtm
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    display_name = fields.Char(
+        automatic=True,
+        compute="_compute_display_name",
+        search="_search_display_name",
+    )
+
+    def _search_display_name(self, operator, value):
+        recs = self.search([]).filtered_domain([("display_name", operator, value)])
```

Comment:
```suggestion
         recs = self.search([("display_name", operator, value)])
```

Reply:
The field display_name is not stored so it is not seachable.